Building failing in arm cortex-m0

Building is failing when `CONFIG_NO_OPTIMIZATIONS` is enabled. This is the output for `./samples/hello_world/` with bbc-microbit as target.

```
Memory region         Used Size  Region Size  %age Used
           FLASH:      103636 B       256 KB     39.53%
            SRAM:        9240 B        16 KB     56.40%
        IDT_LIST:         120 B         2 KB      5.86zephyr/arch/arm/core/libarch__arm__core.a(isr_wrapper.S.obj): In function `_idle_state_cleared':
./zephyr/arch/arm/core/isr_wrapper.S:132:(.text._isr_wrapper+0x2c): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `_IntExit' defined in .text._HandlerModeExit section in zephyr/arch/arm/core/libarch__arm__core.a(exc_exit.S.obj)                                                                                                                             
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
```

--- a/arch/arm/core/isr_wrapper.S
+++ b/arch/arm/core/isr_wrapper.S
@@ -132,5 +132,8 @@ _idle_state_cleared:
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 
-	/* exception return is done in _IntExit() */
-	b _IntExit
+	/* Use 'bx' instead of 'b' because 'bx' can jump further, and use
+	 * 'bx' instead of 'blx' because exception return is done in
+	 * _IntExit() */
+	ldr r0, =_IntExit
+	bx r0

